In [24]:
import numpy as np
from math import factorial

**Задача №1.** Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean):
- среднее арифметическое, <br>
- среднее квадратичное отклонение, <br>
- смещенную и несмещенную оценки дисперсий для данной выборки.

**1. Среднее арифметическое из выборки выпускников** <br>
Определим по следующей формуле:
$$ M(X) = \frac{1}{n}\sum_{i=1}^n x_i $$

In [2]:
array= np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [3]:
avg_array = sum(array)/len(array)

In [4]:
avg_array

65.3

**Ответ**: 65.3

**2. Среднее квадратичное отклонение для данной выборки**<br>
Определим по следующей формуле:<br>
$$ \sigma = \sqrt{\frac{\sum_{i=1}^n (x_i - \bar{x})^2}{n}} $$

In [18]:
array_mean_square_deviation = round(np.sqrt(((array - avg_array) ** 2).sum() / len(array)), 2)

In [19]:
array_mean_square_deviation

30.82

**Ответ**: 30.82

**3. Cмещенная и несмещенная оценка дисперсий для данной выборки**<br>
Определим по следующим формулам:<br>
смещенная дисперсия:
$$ S^2 = \frac{\sum_{i=1}^{n} (x_i - \bar{x})^2}{n} $$
несмещенная дисперсия:  
$$ S^2 = \frac{\sum_{i=1}^{n} (x_i - \bar{x})^2}{n-1} $$

**смещенная дисперсия:**

In [21]:
array_mixed_variance = ((array - avg_array) ** 2).sum() / len(array)

In [22]:
array_mixed_variance

950.11

**Ответ**: 950.11

**несмещенная дисперсия:**

In [35]:
array_unmixed_variance = round(((array - avg_array) ** 2).sum() / (len(array)-1), 2)

In [36]:
array_unmixed_variance

1000.12

**Ответ**: 1000.12

**Задача №2.** В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Cуществует 3 варианта развития событй:<br>
1. из 1-ого ящика вытащили 2 белых мяча, а из 2-ого ящика — 1 белый и 3 не белых, <br>
2. из 1-ого ящика вытащили 1 белый мяч и 1 не белый, а из 2-ого ящика — 2 белых и 2 не белых, <br>
3. из 1-ого ящика вытащили 0 белых мячей, а из 2-ого ящика — 3 белых и 1 не белый.

Вероятность развития события для каждого из вариантов определим по формуле полной вероятности:<br>
$$P(A) = \displaystyle\sum_{i=1}^{n}P(H_i)P(A|_{H_i})$$

Где вероятность события $A$, которая может произойти с одним из событий $H_i$, равна сумме парных произведений вероятностей каждого из этих событий на соответствующей им условной вероятности наступления события $A$

In [33]:
def argument (n, k):
  return int(factorial(n) / (factorial(k) * factorial(n - k)))

**1 вариант:**

$P_1 = \dfrac{C_5^2}{C_8^2} \cdot \dfrac{C_5^1 \cdot C_7^3}{C_{12}^4} \ $

In [30]:
P1 = round((argument(5, 2)*argument(5, 1)*argument(7, 3))/(argument(8, 2)*argument(12, 4)), 4)

In [31]:
P1

0.1263

**2 вариант:** <br>
$P_2 = \dfrac{C_5^1 \cdot C_3^1}{C_8^2} \cdot \dfrac{C_5^2 \cdot C_7^2}{C_{12}^4} \ $

In [35]:
P2 = round((argument(5, 1)*argument(5, 2)*argument(7, 2)*argument(3, 1))/(argument(8, 2)*argument(12, 4)), 4)

In [36]:
P2

0.2273

**3 вариант:**
$P_3 = \dfrac{C_3^2}{C_8^2} \cdot \dfrac{C_5^3 \cdot C_7^1}{C_{12}^4} \ $

In [38]:
P3 = round((argument(3, 2)*argument(5, 3)*argument(7, 1))/(argument(8, 2)*argument(12, 4)), 4)

In [39]:
P3

0.0152

**Вероятность события (А) определим следующим образом:** <br>
$P(А) = P1 + P2 + P3$

In [40]:
PA=P1+P2+P3

In [41]:
PA

0.3688

**Ответ:** Вероятность события А составляет: 0.3688

**Задача №3.** На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. <br>
Вероятность попадания для первого спортсмена равна 0.9, <br>
для второго — 0.8, <br> для третьего — 0.6. <br>
Найти вероятность того, что выстрел произведен: <br>
a) первым спортсменом, <br>
б) вторым спортсменом, <br>
в) третьим спортсменом.

Вероятность того, что выстрел произведен первым спортсменом, вторым и третьим определим используя Теорему Байеса (формулу Байеса), так как событие (попадание в мишень) уже произошло:

$P(B│A)=\dfrac{P(A│B)*P(B)}{P(A)}$

$A$ - попадание в мишень,<br>
$B_i$ - стрелок (1й, 2й, 3й)

$P(B_i|A) = \dfrac{P(A|B_i) \cdot P(B_i)}{P(A)}$

Вероятности выстрела каждым спорстменом равны, так как выстрел один, а спортсменов три. <br>
Вероятность будет равна P(B) = P(B1) = P(B2) = P(B3) = 1/3


In [42]:
P_B = 1/3


Вероятность наступления события A:

In [45]:
P_A = round((P_B*0.9 + P_B*0.8 + P_B*0.6), 4)

In [46]:
P_A

0.7667

**Определим вероятность того, что выстрел произведён одним из спортсменов**

**1-ый спортсмен**:

In [49]:
P1 = round((P_B * 0.9 /P_A), 4)

In [50]:
P1

0.3913

**Ответ:** 0.3913

**2-ой спортсмен**:

In [51]:
P2 = round((P_B * 0.8 /P_A), 4)

In [52]:
P2

0.3478

**Ответ:** 0.3478

**3-ий спортсмен**:

In [53]:
P3 = round((P_B * 0.6 /P_A), 4)

In [54]:
P3

0.2609

**Ответ:** 0.2609

**Проверка:**

In [71]:
P=P1+P2+P3

In [72]:
P

1.0

**Задача №4.** В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. <br>
Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8.<br> Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. <br>
Студент сдал первую сессию. Какова вероятность, что он учится: <br>
a) на факультете A, <br>
б) на факультете B, <br>
в) на факультете C?

Для определения вероятности, что струдент учится на факультете А, В или С определим используя Теорему Байеса (формулу Байеса):

$P(B│A)=\dfrac{P(A│B)*P(B)}{P(A)}$

$A$ - сдача сессии
$B_i$ - факультет студента

$P(B_i|A) = \dfrac{P(A|B_i) \cdot P(B_i)}{P(A)}$




Примем, что: <br>
n - студентов учится на факультете A и В,  
2 * n - студентов учится на факультете C,
B1 - сессию сдал студент с факультета A  
B2 - сессию сдал студент с факультета B  
B3 - сессию сдал студент с факультета C  
А - событие сессия сдана  
События B1, B2, B3 образуют полную группу событий  
P(B1) + P(B2) + P(B3) = 1  
1/n  + 1/n + 1/2n = 1  
n = 2.5  

In [57]:
n=2.5

Вероятность наступления события A:

In [63]:
P_A = round(((1/n)*0.8 + (1/n)*0.7 + (1/(2*n))*0.9), 4)

In [66]:
P_A

0.78

**По формуле Байеса найдем вероятность того на каком факультете учится студент сдавший сессию**

**Факультет А**

In [69]:
 P_1= round(((1/n)*0.8/P_A), 4)

In [70]:
 P_1

0.4103

**Ответ:** 0.4103

**Факультет В**

In [73]:
P_2= round(((1/n)*0.7/P_A), 4)

In [74]:
P_2

0.359

**Ответ:** 0.359

**Факультет С**

In [81]:
P_3= round((((1/(2*n)))*0.9/P_A), 4)

In [82]:
P_3

0.2308

**Ответ:** 0.2308

**Проверка:**

In [85]:
P=round((P_1+P_2+P_3), 1)

In [86]:
P

1.0

**Задача №5.** Устройство состоит из трех деталей. <br> Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. <br> Какова вероятность того, что в первый месяц выйдут из строя:<br>
а) все детали, <br>
б) только две детали, <br>
в) хотя бы одна деталь, <br>
г) от одной до двух деталей?

**Определим вероятность того, что:**


**а) Из строя выходят все детали**


In [90]:
P_1 = round(0.1 * 0.2 * 0.25, 4)

In [91]:
P_1

0.005

**Ответ:** 0.005

**б) Из строя выходят только две детали**


In [92]:
P_2 = round((0.1 * 0.2 * 0.75 + 0.1 * 0.8 * 0.25 + 0.9 * 0.2 * 0.25), 4)


In [93]:
P_2

0.08

**Ответ:** 0.08

**в) Из строя выходит хотя бы одна деталь**

In [96]:
P_3 = round((1 - 0.9 * 0.8 * 0.75), 4)

In [97]:
P_3

0.46

**Ответ:** 0.46

**г) Из строя выходят от одной до двух деталей**


In [103]:
P_4 = round((P_3 - P_1), 4)

In [104]:
P_4

0.455

**Ответ:** 0.455